### PR TITLE
SEARCH-598 and SEARCH-611 (Deleting data folder on Electron Crash)

### PR DESCRIPTION
**Pull request by keerthi16**
## Description
Deleting data folder on Electron Crash

## Approach
How does this change address the problem?

When the electron app is closed or the page is navigated to any other page the data folder gets removed.
But if the application crashes the data folder won't be deleted. 
And this PR handles deleting the data folder when it's crashed.

[SEARCH-598](https://perzoinc.atlassian.net/browse/SEARCH-598)
### Implementation steps: 

-- Cron job
1. When the electron app is opened, I am creating a `LaunchAgent` file by passing the PID of the application.
2. Every (set time) 10 Min the `LaunchAgent` file executes and checks for the PID process exist or not.
3. If exists, It won't does anything.
4. If not, Its deletes the data folder and stop the "LaunchAgent".

-- On Login
1. Same LaunchAgent method but different script deletes the folder on user login

[SEARCH-611](https://perzoinc.atlassian.net/browse/SEARCH-611)
### Implementation steps: 
-- Cron job
1. Windows as same steps as Mac but using the `Task Scheduler`

-- On Login
Created windows registry for deleting the data folder on user login

## Learning
Task Scheduler --> [Schtasks.exe](https://msdn.microsoft.com/en-us/library/windows/desktop/bb736357(v=vs.85).aspx)
LaunchAgents --> [Doc](https://developer.apple.com/library/content/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLaunchdJobs.html)

#### Blog Posts
N/A

## Related PRs
N/A

Jenkins build passing --> #21

_Note: Mac implementation is blocked by [ELECTRON-295](https://perzoinc.atlassian.net/browse/ELECTRON-295)_

@VikasShashidhar @serkanmulayim  Please review. Thanks

## Open Questions if any and Todos
- [x] Unit-Tests
[Test Results — Unit Tests.pdf](https://github.com/symphonyoss/SymphonyElectron/files/1814926/Test.Results.Unit.Tests.pdf)
[Test Results — Spectron.pdf](https://github.com/symphonyoss/SymphonyElectron/files/1814927/Test.Results.Spectron.pdf)
- [x] Documentation
- [ ] Automation-Tests

### CODE REVIEW

File: installer/win/Symphony-x64.aip [hunk 1]
```diff
@@ -124,6 +124,7 @@
     <ROW Component="appupdate.yml" ComponentId="{F7586760-660A-4C38-8937-138DBEC18D34}" Directory_="resources_Dir" Attributes="0" KeyPath="appupdate.yml" Type="0"/>
     <ROW Component="blink_image_resources_200_percent.pak" ComponentId="{56AB17A5-B690-4CBE-A39D-512381AAAFE1}" Directory_="APPDIR" Attributes="0" KeyPath="blink_image_resources_200_percent.pak" Type="0"/>
     <ROW Component="cld.node" ComponentId="{D9F74D7B-B2E4-4ED8-A023-A54A98AE3F96}" Directory_="Release_Dir" Attributes="256" KeyPath="cld.node" Type="0"/>
+    <ROW Component="clearonlaunch.exe" ComponentId="{8429AC5F-4C57-4DAD-BEDF-B7FCC0714DB2}" Directory_="library_Dir" Attributes="256" KeyPath="clearonlaunch.exe"/>
```
Comment (VikasShashidhar) [inline installer/win/Symphony-x64.aip]: Can you change this to `symphony-clear-on-launch.exe`?

File: installer/win/Symphony-x64.aip [hunk 2]
```diff
@@ -140,13 +141,14 @@
     <ROW Component="node.dll" ComponentId="{C0972355-339E-438C-94A3-74174DE4C6B6}" Directory_="APPDIR" Attributes="256" KeyPath="node.dll"/>
     <ROW Component="node_modules" ComponentId="{A4EB33A8-FEA8-40A5-94EF-705EBE64DDC1}" Directory_="node_modules_Dir" Attributes="0"/>
     <ROW Component="npmignore" ComponentId="{849CFFE2-EBC3-4430-AC8B-DEE1B66DF589}" Directory_="spawnrx_Dir" Attributes="0" KeyPath="npmignore" Type="0"/>
+    <ROW Component="searchwintask.exe" ComponentId="{BF26B582-706C-4464-8F4C-DD2AB21A44EC}" Directory_="library_Dir" Attributes="256" KeyPath="searchwintask.exe"/>
```
Comment (VikasShashidhar) [inline installer/win/Symphony-x64.aip]: Can you rename this to a more appropriate name?

File: installer/win/Symphony-x64.aip [hunk 3]
```diff
@@ -210,6 +212,7 @@
     <ROW File="build.sh" Component_="npmignore" FileName="build.sh" Attributes="0" SourcePath="..\..\dist\win-unpacked\resources\app.asar.unpacked\node_modules\spawn-rx\build.sh" SelfReg="false" NextFile="CODE_OF_CONDUCT.md"/>
     <ROW File="ca.pak" Component_="am.pak" FileName="ca.pak" Attributes="0" SourcePath="..\..\dist\win-unpacked\locales\ca.pak" SelfReg="false" NextFile="cs.pak"/>
     <ROW File="cld.node" Component_="cld.node" FileName="CLD~1.NOD|cld.node" Attributes="0" SourcePath="..\..\dist\win-unpacked\resources\app.asar.unpacked\node_modules\@paulcbetts\cld\build\Release\cld.node" SelfReg="false" NextFile="indexvalidatorx64.exe"/>
+    <ROW File="clearonlaunch.exe" Component_="clearonlaunch.exe" FileName="CLEAR-~1.EXE|clear-on-launch.exe" Attributes="0" SourcePath="..\..\library\clear-on-launch.exe" SelfReg="false" NextFile="searchwintask.exe" DigSign="true"/>
```
Comment (VikasShashidhar) [inline installer/win/Symphony-x64.aip]: Same here.

File: installer/win/Symphony-x64.aip [hunk 4]
```diff
@@ -271,14 +274,15 @@
     <ROW File="ptPT.pak" Component_="am.pak" FileName="pt-PT.pak" Attributes="0" SourcePath="..\..\dist\win-unpacked\locales\pt-PT.pak" SelfReg="false" NextFile="ro.pak"/>
     <ROW File="ro.pak" Component_="am.pak" FileName="ro.pak" Attributes="0" SourcePath="..\..\dist\win-unpacked\locales\ro.pak" SelfReg="false" NextFile="ru.pak"/>
     <ROW File="ru.pak" Component_="am.pak" FileName="ru.pak" Attributes="0" SourcePath="..\..\dist\win-unpacked\locales\ru.pak" SelfReg="false" NextFile="sk.pak"/>
+    <ROW File="searchwintask.exe" Component_="searchwintask.exe" FileName="SEARCH~1.EXE|search-win-task.exe" Attributes="0" SourcePath="..\..\library\search-win-task.exe" SelfReg="false" DigSign="true"/>
```
Comment (VikasShashidhar) [inline installer/win/Symphony-x64.aip]: Same here.

File: js/search/searchConfig.js [hunk 5]
```diff
@@ -26,12 +26,27 @@ const libraryPath = isMac ? path.join(macLibraryPath, 'libsymphonysearch.dylib')
...
 const userConfigFile = isDevEnv ? path.join(__dirname, '..', '..', userConfigFileName) : path.join(userData, userConfigFileName);
 
+const libraryFolderPath = isMac ? macLibraryPath : winLibraryPath;
+
+const launchAgentFile = path.join(libraryFolderPath, 'search-launch-agent.sh');
+const launchDaemonFile = path.join(libraryFolderPath, 'search-launch-daemon.sh');
+const windowsTaskFile = path.join(libraryFolderPath, 'search-win-task.exe');
+const windowsClearScript = path.join(libraryFolderPath, 'clear-on-launch.exe');
+
+const scriptExt = isMac ? '.sh' : '.exe';
```
Comment (VikasShashidhar) [inline js/search/searchConfig.js]: It's a little dangerous to assume that it has to be an "exe" extension if it is not a Mac. What if we decide to support Linux in the future?

File: js/search/utils/search-launchd.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+const logLevels = require('../../enums/logLevels.js');
+const Winreg = require('winreg');
+
+/**
+ * Register for creating launch agent
+ * @type {Registry}
+ */
+const regKey = new Winreg({
+    hive: Winreg.HKCU,
+    key: '\\Software\\Microsoft\\Windows\\CurrentVersion\\Run'
```
Comment (VikasShashidhar) [inline js/search/utils/search-launchd.js]: Is this the right location in the registry?

Comment (VikasShashidhar) [reply js/search/utils/search-launchd.js]: Also, is there no way to avoid using the registry?

Comment (keerthi16) [reply js/search/utils/search-launchd.js]: Yes, the location is correct.
No, this is the only option.

File: js/search/searchConfig.js (hunk 5 above)
Comment (keerthi16) [reply js/search/searchConfig.js]: 👍

**Comment by VikasShashidhar** [comment]
@keerthi16 Can you please resolve conflicts?

**Comment by keerthi16** [comment]
@VikasShashidhar conflicts resolved :)

**Comment by VikasShashidhar** [comment]
@VishwasShashidhar Can you review and merge this PR?

**Pull request merged**